pydantic/env_settings.py

```diff
@@ -76,7 +76,9 @@ class Config:
         @classmethod
         def prepare_field(cls, field: ModelField) -> None:
             env_names: Union[List[str], AbstractSet[str]]
-            env = field.field_info.extra.get('env')
+            field_info_from_config = cls.get_field_info(field.name)
+
+            env = field_info_from_config.get('env') or field.field_info.extra.get('env')
```

Comment:
I'm not entirely clear what this order of priority means in practice.

E.g. if we set `env` on the parent and the child, does the child take priority? Does the same apply both when done via config and done using `Field(..., env='my_api_key')`?

Do we need to add documentation about this? We definitely need to add tests to confirm and lock the behaviour is as expected.

Reply:
So, child should definitely override the parent with the child config, and as far as I understand, parent declaration method is irrelevant, because, by the time of this statement, the configuration in the parent is already applied and expressed in `field.field_info.extra.get('env')
`

> Does the same apply both when done via config and done using Field(..., env='my_api_key')?

When declaring the same field in the child with `Field(`, you effectively override the whole field. 

I added test to confirm the behavior I expected, could you specify what other scenarios you want to get tested?

Reply:
can we add a test for something like

```py
def test_env_inheritance_config(env):
    class Parent(BaseSettings):
        a: str = Field(None, env='a_in_field')
        b: str = None
        c: str = None
        d: str = None

        class Config:
            fields = {
                'b': {'env': ['b_in_config']},
                'c': {'env': ['c_in_config']},
            }

    class Child(Parent):
        class Config:
            env_prefix = 'prefix_'
            fields = {
                'a': {'env': ['a_child_config']},
                'b': {'env': ['b_child_config']},
            }

    env.set('a_in_field', 'a_in_field')
    env.set('b_in_config', 'b_in_config')
    env.set('c_in_config', 'c_in_config')
    env.set('d', 'd')
    env.set('a_child_config', 'a_child_config')
    env.set('b_child_config', 'b_child_config')
    env.set('prefix_d', 'prefix_d')

    assert Child().dict() == ...
```
I'm just not 100% sure what would happen in each case, so best to add a test to be explicit.

Reply:
maybe it's obvious and the test does exactly as expected, in which case you could consider it unnecessary, but since even I'm not exactly clear what would happen, better to add the test.

Reply:
Considering your snippet, I'd rather break it down into individual cases and name them correspondingly, this way it will be more readable and it is less likely that we 've missed something